### DISCUSSION TITLE
Can't use detectron2 even after installing.

### DISCUSSION THREAD

**Discussion by cooldecola** [facebookresearch/detectron2#3462] [Q&A] [answered]
I'm trying to run a simple program based on the detectron2 collab tutorial. I'm fairly new to detectron2, before this I only used detectron (so I have cuda and cudnn already setup). I've followed the instructions on the website many times, however when I try running any code that imports detectron2 , it says "detectron2 is not a package" 

### My Code
This is the code I'm trying to run (my editor is VSCode): 

```
import torch, torchvision
print(torch.__version__, torch.cude.is_available())
assert torch.__version__.startswith("1.9")

import detectron2
from detectron2.utils.logger import setup_logger
```

### Error
However I get the following error when I run `python detectron2.py`  on terminal:
```
File "/home/Repo/detectron2_practice/detectron2.py", line 6, in <module> 
from detectron2.utils.logger import setup_logger
ModuleNotFoundError: No module named 'detectron2.utils'; 'detectron2' is not a package
```

### Specs
These are my specs:
- Ubuntu 18.04.5 LTS
- Python 3.6
- g++ 7.5.0
- gcc 7.5.0
- nvcc: 10.2, V10.2.89
pip3 list: 
- torch 1.9.0+cu102
- torchvision 0.10.0+cu102
- torchaudio 0.8.0
- detectron2 0.5+cu102 (/home/.local/python3.6/site-packages)
 
### What I did
- I followed the instructions on the installation page. So first I installed pytorch and torchvision that corresponds to my cuda 10.2:
`pip install torch==1.8.0 torchvision==0.9.0 torchaudio==0.8.0`

- Then I install pre-built Detectron2 (linux only) (cuda 10.2, torch 1.8)
```
python -m pip install detectron2 -f \
  https://dl.fbaipublicfiles.com/detectron2/wheels/cu102/torch1.8/index.html
```

### Additional Notes
- On VSCode when I type in import detectron2 I can see the autocomplete options and further options like "config, model_zoo, layers.. etc". 
- When on terminal I enter python3 interactive shell and type "import detectron2" , i Get no errors. 
- The problem only arises when I try to run my python file. 

Any help in this would be phenomenal, thanks!

**Comment by ppwwyyxx** [comment, marked as answer]
You cannot name your own file `detectron2.py` if you want to import the detectron2 library.

**Reply by cooldecola** [reply]
Such a dumb error! But thank you so much for helping
